### PR TITLE
config: allow mon description for `monitor` value

### LINKED ISSUE #137: [REQ] Using a monitor's description for monitor value
If I try to do this:
```
input-field {
  monitor = desc:GIGA-BYTE TECHNOLOGY CO. LTD. G27QC 0x00000B1D
```

it won't actually put the input-field on that monitor.

This exact monitor value works on Hyprland

### PR DESCRIPTION
**Pull request by bvr-yr**
that's it? or should it be `desc:`

closes #137

### CODE REVIEW

File: src/renderer/Renderer.cpp [hunk 1]
```diff
@@ -306,7 +306,7 @@ std::vector<std::unique_ptr<IWidget>>* CRenderer::getOrCreateWidgetsFor(const CS
         const auto CWIDGETS = g_pConfigManager->getWidgetConfigs();
 
         for (auto& c : CWIDGETS) {
-            if (!c.monitor.empty() && c.monitor != surf->output->stringPort)
+            if (!c.monitor.empty() && c.monitor != surf->output->stringPort && surf->output->description.find(c.monitor) != 0)
```
Comment (vaxerski) [inline src/renderer/Renderer.cpp:309]: my brother in christ `.starts_with`

File: src/renderer/AsyncResourceGatherer.cpp [hunk 2]
```diff
@@ -42,7 +42,8 @@ CAsyncResourceGatherer::CAsyncResourceGatherer() {
     }
 
     for (auto& mon : mons) {
-        const auto MON = std::find_if(g_pHyprlock->m_vOutputs.begin(), g_pHyprlock->m_vOutputs.end(), [mon](const auto& other) { return other->stringPort == mon; });
+        const auto MON = std::find_if(g_pHyprlock->m_vOutputs.begin(), g_pHyprlock->m_vOutputs.end(),
+                                      [mon](const auto& other) { return other->stringPort == mon || other->description.find(mon) == 0; });
```
Comment (vaxerski) [inline src/renderer/AsyncResourceGatherer.cpp:46]: you have to check for empty desc

also how the fuck are you comparing a pointer to a string???

Comment (bvr-yr) [reply src/renderer/AsyncResourceGatherer.cpp:46]: ehm, I didn't even change this.
isn't `mon` a member of vector of strings? (from widget conf) [here](https://github.com/hyprwm/hyprlock/pull/216/files#diff-7cc329661269fc3fc040b387dac691ae82028fce0622b565340fc08274047c33R24)
205854-area- 971 607-634x179

Comment (bvr-yr) [reply src/renderer/AsyncResourceGatherer.cpp:46]: or do you mean description field? it's string [here](https://github.com/hyprwm/hyprlock/pull/216/files#diff-588e4c1eeffc3faee9f156543179767a6a4c8f352637af7d3e0bfb7d7484185bR50)
should've named it different

Comment (vaxerski) [reply src/renderer/AsyncResourceGatherer.cpp]: no I mean `mon`

Comment (bvr-yr) [reply src/renderer/AsyncResourceGatherer.cpp]: the same like it is and was compared to `stringPort`? I didn't even touch this part. Or I can't get you.
isn't `mon` a member of vector of strings (from widgets conf) [here](https://github.com/hyprwm/hyprlock/pull/216/files#diff-7cc329661269fc3fc040b387dac691ae82028fce0622b565340fc08274047c33R24)?

184107-area- 117 371-904x176

184250-area- 719 447-766x182

Comment (vaxerski) [reply src/renderer/AsyncResourceGatherer.cpp]: oh god am dum

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**